test/miscellaneous_tests/nonlinear_solve.jl

```diff
@@ -20,67 +23,68 @@ let
     end
     
     # Creates NonlinearProblem.
-    u0 = rand(rng, length(unknowns(steady_state_network_1)))
-    p = rand(rng, length(parameters(steady_state_network_1)))
-    nl_prob = NonlinearProblem(steady_state_network_1, u0, p)
+    u0 = rnd_u0(steady_state_network_1, rng)
+    ps = rnd_ps(steady_state_network_1, rng)
+    nlprob = NonlinearProblem(steady_state_network_1, u0, ps)
     
     # Solves it using standard algorithm and simulation based algorithm.
-    sol1 = solve(nl_prob; abstol=1e-12, reltol=1e-12).u
-    sol2 = solve(nl_prob, DynamicSS(Rosenbrock23()); abstol=1e-12, reltol=1e-12).u
+    sol1 = solve(nlprob; abstol=1e-12, reltol=1e-12)
+    sol2 = solve(nlprob, DynamicSS(Rosenbrock23()); abstol=1e-12, reltol=1e-12)
     
     # Tests solutions are correct.
-    @test isapprox(sol1[1], p[1] / p[2]; atol=1e-10)
-    @test isapprox(sol1[2]^3 / factorial(3), p[3] / p[4]; atol=1e-10)
-    @test isapprox(sol1[3], (p[5] * p[2]) / (p[6] * p[1]); atol=1e-10)
-    @test isapprox(sol2[1], p[1] / p[2]; atol=1e-10)
-    @test isapprox(sol2[2]^3 / factorial(3), p[3] / p[4]; atol=1e-10)
-    @test isapprox(sol2[3], (p[5] * p[2]) / (p[6] * p[1]); atol=1e-10)
+    @test sol1[:X1] ≈ nlprob.ps[:k1] / nlprob.ps[:k2] atol=1e-10
```

Comment:
Uses symbolic (instead of integer) indexing.

Loads of test where also written before I was familiar with nice ways to use `≈`, `@test`, and `atol`, so I rewrote quite a few tests using this while I was at it.